application/blueprints/datamanager/services/endpoint.py

```
@@ -34,9 +36,57 @@ def get_endpoint_urls_for_hashes(hashes: list) -> dict:
+
+    datasette_url = current_app.config.get("DATASETTE_BASE_URL")
+    hash_list = ",".join(f"'{h}'" for h in hashes)
+    sql = (
+        "SELECT l.endpoint AS endpoint, "
+        "l.status AS latest_status, "
+        "l.entry_date AS latest_log_entry_date, "
+        "(SELECT max(l2.entry_date) FROM log l2 "
+        "WHERE l2.endpoint = l.endpoint AND l2.status = '200') AS latest_200_date "
+        "FROM log l "
+        "WHERE l.rowid IN ("
+        f"SELECT max(rowid) FROM log WHERE endpoint IN ({hash_list}) GROUP BY endpoint)"
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)